### PR TITLE
HOUSNAV-165: Add Horizontal Pod Autoscaler (HPA)

### PR DESCRIPTION
**Pull request by lazyniuk**
[HOUSNAV-165](https://hous-bssb.atlassian.net/browse/HOUSNAV-165)

## Overview & Purpose

By implementing HPA, we aim to optimize the resource allocation of our pods by automatically adjusting CPU and memory requests based on real-time usage patterns. 

## Summary of Changes

Added Horizontal Pod Autoscaler (HPA) configuration: Implemented HPA to automatically manage CPU and memory requests for pods in the Openshift project.

## Testing

Tested on *-tools namespace

## Checklist
- [ ] I have written or updated vitests for this work
- [ ] I have reviewed the [accessibility guidelines](https://digital.gov.bc.ca/wcag/home/) relevant to this work
- [ ] I have reviewed this work with at least one screen reader (when applicable)
- [ ] I have added/updated any related documentation

### CODE REVIEW

File: charts/housnav/values-uat.yaml [hunk 1]
```diff
@@ -10,11 +10,11 @@ readinessProbe:
 
 resources:
   requests:
-    memory: "50Mi"
-    cpu: "25m"
-  limits:
     memory: "100Mi"
     cpu: "50m"
+  limits:
```
Comment (jbook24) [inline charts/housnav/values-uat.yaml:15]: What inspired the change here?

**Comment by jbook24** [comment]
What was the documented consideration against using HPA?

**Comment by lazyniuk** [comment]
> What was the documented consideration against using HPA?

Great comment! After running load tests with Apache JMeter, I decided that HPA is a better choice than VPA for our project. So, I reconfigured the autoscaling to use HPA.

File: charts/housnav/values-uat.yaml (hunk 1 above)
Comment (lazyniuk) [reply charts/housnav/values-uat.yaml:15]: > What inspired the change here?

I did a more accurate load test with Apache JMeter and found the optimal value for the moment. These values take into account the limits we have on this project.

**Review by jbook24 — APPROVED** [review]

**Pull request merged**